runtime/src/slots.rs

```diff
@@ -155,22 +158,46 @@ decl_storage! {
 
 		/// Amounts currently reserved in the accounts of the bidders currently winning
 		/// (sub-)ranges.
-		pub ReservedAmounts get(reserved_amounts): map Bidder<T::AccountId, ParaIdOf<T>> => Option<BalanceOf<T>>;
+		pub ReservedAmounts get(reserved_amounts): map Bidder<T::AccountId> => Option<BalanceOf<T>>;
 
 		/// The set of Para IDs that have won and need to be on-boarded at an upcoming lease-period.
 		/// This is cleared out on the first block of the lease period.
-		pub OnboardQueue get(onboard_queue): map LeasePeriodOf<T> => Vec<ParaIdOf<T>>;
+		pub OnboardQueue get(onboard_queue): map LeasePeriodOf<T> => Vec<ParaId>;
 
 		/// The actual on-boarding information. Only exists when one of the following is true:
 		/// - It is before the lease period that the parachain should be on-boarded.
 		/// - The full on-boarding information has not yet been provided and the parachain is not
 		/// yet due to be off-boarded.
-		pub Onboarding get(onboarding): map ParaIdOf<T> =>
+		pub Onboarding get(onboarding): map ParaId =>
 			Option<(LeasePeriodOf<T>, IncomingParachain<T::AccountId, T::Hash>)>;
 
 		/// Off-boarding account; currency held on deposit for the parachain gets placed here if the
 		/// parachain gets off-boarded; i.e. its lease period is up and it isn't renewed.
-		pub Offboarding get(offboarding): map ParaIdOf<T> => T::AccountId;
+		pub Offboarding get(offboarding): map ParaId => T::AccountId;
+	}
+}
+
+impl<T: Trait> OnSwap for Module<T> {
+	fn can_swap(one: ParaId, other: ParaId) -> Result<(), &'static str> {
```

Comment:
Returning `bool` on a `can_*` function would be more idiomatic, no?
```suggestion
	fn can_swap(one: ParaId, other: ParaId) -> bool {
		!<Onboarding<T>>::exists(one) && !<Onboarding<T>>::exists(other)
	}
```